### DISCUSSION TITLE
categories 테이블의 정규화 이슈

### DISCUSSION THREAD

**Discussion by newnyee** [newnyee/refactoring-habit#14] [Q&A] [answered]
categories 테이블의 정규화 관련 이슈에 대해 말씀드립니다.

1. categories 대분류 속성에 따른 image 필요
![image](https://github.com/newnyee/refactoring-habit/assets/121937711/c07ffa31-961e-4999-aa2d-a9a7865254ef)
위의 이미지처럼 대분류에 해당되는 이미지가 필요하여 image라는 속성이 추가되어야할 것 같습니다.

2. 그에 따른 categories 값 저장 결과 
![image](https://github.com/newnyee/refactoring-habit/assets/121937711/09403b88-cab4-4209-b800-f2845ae685b6)

  (예시 이미지)

  categories 테이블의 구조상 위의 예시 이미지 처럼 large, image 속성의 카디널리티가 낮습니다.
따라서 대분류와 중분류 테이블로 나누는 정규화 과정이 있어야할지, 아니면 지금처럼 정규화가 덜 된 상태로 사용하는게 좋을 지 고민입니다.

3. 한글? 영어?
카테고리 테이블 저장시 카테고리 네임을 어떤 언어로 저장할 지 고민됩니다.
(영어와 한글 둘 다 사용할 지, 아니면 영어만 사용할 지)
영어는 기본적으로 포함 시키려는 이유는 url 상으로 보여지기 때문입니다.

**Comment by taejin-chang** [comment, marked as answer]
- 정규화가 된 상태로 구현하는 테이블을 category_large 와 category_middle로 분리하기로 결정됨.
- 카테고리에서 사용되는 한글명칭과 영문명칭이 UI에서 혼재되어 사용될 예정이기에 국제화를 고려한 리소스파일로 분리할 생각도 해보았으나, 관리자가 카테고리를 직접 등록해서 사용가능 여부 또한 제어할 예정이기 때문에 테이블로 관리하도록 결정됨.
- 관리자가 등록한 카테고리의 사용가능 여부를 결정하는 status 컬럼을 category_large에 추가하기로 결정됨.
- images 컬럼은 category_large에 포함시키도록 결정됨.

**Reply by newnyee** [reply]
![image](https://github.com/newnyee/refactoring-habit/assets/121937711/7a1b4e00-1ef1-4261-867e-d21fbefc8b3d)
정규화 완료
